site/app/libraries/routers/WebRouter.php

```diff
@@ -55,17 +61,22 @@ public function __construct(Request $request, Core $core, $logged_in) {
     }
 
     public function run() {
-        $controllerName = $this->parameters['_controller'];
-        $methodName = $this->parameters['_method'];
-        $controller = new $controllerName($this->core);
+        $this->controller_name = $this->parameters['_controller'];
+        $this->method_name = $this->parameters['_method'];
+        $controller = new $this->controller_name($this->core);
 
         foreach ($this->parameters as $key => $value) {
             if (Utils::startsWith($key, "_")) {
                 unset($this->parameters[$key]);
             }
         }
 
-        return call_user_func_array(array($controller, $methodName), $this->parameters);
+        // pass $_GET to controllers
+        // the user-specified $_GET should NOT override the controller name and method name matched
+        $this->request->query->remove('url');
```

Comment:
If we go to `http://192.168.56.111`, the `url` would be empty, but no need to worry as the `remove` function calls `unset`, so it will be fine if `url` does not exist in the query.